build.sbt

```diff
@@ -1479,11 +1473,20 @@ lazy val runtime = (project in file("engine/runtime"))
         (Benchmark / testOnly).toTask(" -- -z " + name).value
       }
     }.evaluated,
-    Benchmark / parallelExecution := false
   )
+  /** Benchmark settings  */
   .settings(
+    inConfig(Benchmark)(Defaults.testSettings),
+    Benchmark / javacOptions --= Seq(
+      "-source",
+      frgaalSourceLevel,
+      "--enable-preview"
+    ),
     (Benchmark / javaOptions) :=
-      (LocalProject("std-benchmarks") / Benchmark / run / javaOptions).value
+      (LocalProject("std-benchmarks") / Benchmark / run / javaOptions).value,
+    (Benchmark / javaOptions) ++= benchOnlyOptions,
```

Comment:
This is what was missing before - `benchOnlyOptions` forwards `-Dbench.compileOnly` system property to the forked bench runner.